components/kyma-environment-broker/internal/edp/client_fake.go

```diff
@@ -0,0 +1,79 @@
+package edp
+
+import "errors"
+
+// FakeClient implements the edp client interface but does not process data nor call real external system
+type FakeClient struct {
+	data map[string]interface{}
```

Comment:
you can define the struct as a type then use it in the map

Reply:
Imagine that I'd like to use the fake client and create two different DataTenants. Will it work? I mean can I assert it at the end?

Reply:
Good point, we will not be able to check the fields of map value since it is an anonymous struct.